src/main/java/com/hotels/bdp/circustrain/bigquery/client/BigQueryMetastoreClientFactory.java

```diff
@@ -37,7 +37,7 @@
       BigQueryMetastore bigQueryMetastore,
       ExtractionService service,
       TableServiceFactory tableServiceFactory) {
-    this.metastoreClient = new BigQueryMetastoreClient(bigQueryMetastore, service, new HiveTableCache(),
+    metastoreClient = new BigQueryMetastoreClient(bigQueryMetastore, service, new HiveTableCache(),
```

Comment:
Maybe one for a future refactoring but it feels like the client could create its own cache instead of being passed one. Less coupling of that cache between classes. In future we maybe even make it a singleton but let's not make the scope of this PR too wide now.